src/utils.ts

```diff
@@ -82,3 +83,36 @@ export function doPatternsMatchUrl (remotePattern: RemotePattern, parsedUrl: Par
 
   return true
 }
+
+export class Lock {
```

Comment:
sidenote, but I wonder if we could use the `Lock` class elsewhere outside of this project. 🤔